React/nested-event-handling.md

```diff
@@ -180,187 +197,103 @@ The CSS is the most straightforward thing on our todo list, so let's get that ou
-
-Let's create a new function in the `StudentCollection` component. This function should take in one parameter which represents the unique identifier for the `<Student />` whom we want to mark "present".
-
-```javascript
-//StudentCollection.js
-
-markPresent = (studentIndex) => {
-  console.log(studentIndex);
-}
-```
-
-
-<details>
-<summary><b>Question:</b>  What way do we have to pass data from a parent component to a child component.
-</summary>
-If you answered `props`, you're correct!
-
-We are going to send _a callback function_ as a part of `props` from the `StudentCollection` component to the `Student` component. Once the event occurs in the child component, we can then get the data we need from the child, and pass that to the parent to update the `state`.
-</details>
-
-Update the `render` function in the `StudentCollection` component to pass in a new `prop` to `Student`, the callback function. Note: We are _passing_ this as a function not _calling_ the function, so we leave the parenthesis off.
-
-```javascript
-// StudentCollection.js
-this.state.students.map((student, i) => {
-  return (
-    <li key={ i }>
-      <Student
-        index={ i }
-        fullName={ student.fullName }
-        email={ student.email }
-        isPresent={ student.isPresent }
-        markPresentCallback={ this.markPresent }
-        />
-    </li>
-  );
-});
-```
-
-Now, in the `Student` component, we can call this callback function within our event handler with the information about the specific student that needs to be updated.
-
-```javascript
-// Student.js
-const Student = (props) => {
-  const onPresentButtonClick = () => {
-    console.log(props);
-    props.markPresentCallback(props.index);
-  }
-
-  // The rest of the file
-  // ...
-}
-```
-
-Finally, we'll update the callback function in the parent component to modify the state of the student.
-
-```javascript
-//StudentCollection.js
-
-markPresent = (studentIndex) => {
-  console.log(studentIndex);
-
-  // Store our state in a local variable so we can make the update
-  let updatedStudents = this.state.students;
-  updatedStudents[studentIndex].isPresent = true;
-
-  // Call setState to update our state (as well as re-render automatically)
-  this.setState({ students: updatedStudents });
-}
-```
-
 ### Event Handling Summary
 
 Wow, that was kind of complex. Let's look at a diagram of what's going on.
 
 ![State management with callbacks](images/state-management-with-callbacks.png)
 
-<!-- https://drive.google.com/open?id=1byKvDyUP5HUwQojmg2cpL4eNO9keFYpD -->
+<!-- https://drive.google.com/file/d/1dyNbUUe_6rkrSEG_lPKJGiwAAOGIHQ5_/view?usp=sharing -->
 
 **On the left in orange** is the setup. Information is passed from parent components to child components at `render` time using `props`.
+
+- `App` provides a list of students to `StudentCollection`
 - `StudentCollection` gives `Student`:
```

Comment:
We need to adjust this since there isn't a `StudentCollection` anymore.

Reply:
Actually there is it forwards individual student data to each student component.